apps/oxlint/src/lint.rs

```diff
@@ -280,6 +281,32 @@ impl Runner for LintRunner {
 
         let lint_config = config_builder.build();
 
+        // Enable type-aware linting if:
+        // 1) `--tsconfig` is passed
+        // 2) `tsgolint` appears to be installed
+        let type_aware = basic_options.tsconfig.is_some()
+            && (
+                // Check if `tsgolint` is executable
+                std::process::Command::new("tsgolint").arg("--help").output().is_ok()
```

Comment:
how are we going to find this binary. unless a user has installed it globally, this command will probable fail

Reply:
I outlined a brief plan here: https://github.com/oxc-project/oxc/issues/12455. In short, I think we should check (in order):

1. Check if `$OXLINT_TSGOLINT_PATH` is defined
2. Check if `$CWD/node_modules/.bin/tsgolint` exists
3. Check if `$PATH/tsgolint` exists (<-- essentially what we are doing now)

I didn't include this in the initial PR as I didn't want to complicate the initial implementation for review.